### ISSUE TITLE
HighChart not compatible with CSP

### ISSUE THREAD

**Issue by gbataille** [michelson/lazy_high_charts#144]
Hey guys,

so I'm working on small gem that should help the adoption and the reporting of CSP. (see http://www.w3.org/TR/CSP/).
I wanted to add charts in my views. I chose highcharts and your lib. While I appreciated some aspects of it (you don't see any JS for example), I have a core problem. CSP rightly considers the inline JS as dangerous. and you are doing online JS unfortunately. since my gem is for CSP protected site, this is clearly incompatible. But I wanted to give you a heads up because one could hope that CSP is going to be more and more used in the future.

it might be a good idea for you to propose other integration methods. As an example, here is my commit: https://github.com/gbataille/csp_report/commit/282e4b520dff63c81bab6a8c022a418bdc0d0d5d
Basically I compose the Chart object in the controller. I put it in a div data attribute in the view. In the JS, I intercept this data and use the div as the graph hook.
That is clearly slightly more complex, with more moving pieces, but it is considered a better way from a security perspective and will work with CSP directives on.

I'll let you have a look and see for yourself. Happy to talk about it if you want.

Cheers.
